server/controllers/group_controller.js

```diff
@@ -1,37 +1,37 @@
 import Group from '../models/group';
 
-const ObjectId = require('mongoose').Types.ObjectId;
-
-export const createGroup = (name, description, creator) => {
-  // TODO: error check on names, descriptions and creators?
+export const createGroup = (name, description, userId) => {
   const group = new Group();
   group.name = name;
   group.description = description;
-  group.creator = creator; // TODO: @ploomis how to get authenticated user ??
+  group.creator = userId; // how to get authenticated user ??
+
   group.createDate = Date.now();
   group.editDate = Date.now();
-  group.members.push(ObjectId(creator));   // TODO: make sure creator is a valid user ID
+  group.members.push(userId);   // TODO: make sure creator is a valid user ID
 
   return group.save(); // TODO: catch errors
 };
 
-export const addGroupMember = (groupid, memberid) => {
+export const addGroupMember = (groupIds, userId) => {
 // TODO: similar to addGroupArticle
-  Group.findOne({ _id: ObjectId(groupid) }, (err, group) => {
-    group.members.push(memberid); // TODO: validate member is a real member
-    return group.save();
-  });
+  for (let i = 0; i < groupIds.length; i++) {
+    const groupId = groupIds[i];
+    Group.findByIdAndUpdate(groupId, { $push: { members: userId } });
+  }
 };
 
-export const addGroupArticle = (groupid, articleid) => {
-  Group.findOne({ _id: ObjectId(groupid) }, (err, group) => {
-    // if (err) return; TODO: figure out how to actually catch errors
-    group.articles.push(articleid);
-    return group.save();
-  });
+// can this just be called on an array of groupids? yes
+export const addGroupArticle = (groupIds, articleId) => {
+  for (let i = 0; i < groupIds.length; i++) {
+    const groupId = groupIds[i];
+
+    Group.findByIdAndUpdate(groupId, { $push: { articles: articleId } });
+  }
 };
 
-export const getGroup = (id) => {
-  return Group.findOne({ _id: ObjectId(id) })
-              .populate('articles');
+export const getGroup = (userId, groupId) => {
+  // need to check if user is in the group OR the group is public
+  return Group.findOne({ _id: groupId });
```

Comment:
Yeah. For now I'm just going to change the query, but there should probably be explicit checks? Mostly as an issue of consistency